scripts/doc_checker.py

```diff
@@ -24,97 +40,242 @@
         return None
-    signature = inspect.signature(func)
+    signature = inspect.signature(obj)
     return {
         k: v.default
         for k, v in signature.parameters.items()
         if v.default is not inspect.Parameter.empty
     }
 
 
-def validate_modin_error(import_path):
-    from numpydoc.validate import Docstring
+def check_optional_args(doc: Docstring) -> list:
+    """
+    Check type description of optional arguments.
+
+    Parameters
+    ----------
+    doc : Docstring
+
+    Returns
+    -------
+    list
+        List of tuples with Modin error code and its description.
+    """
+    if not doc.doc_parameters:
+        return []
+    optional_args = get_optional_args(doc)
+    if not optional_args:
+        return []
 
-    doc = Docstring(import_path)
     errors = []
-    default_args = get_default_args(doc.obj)
-    if not default_args:
-        return None
-    for parameter in doc.doc_parameters:
-        if parameter in default_args:
-            type_line = doc.doc_parameters[parameter][0]
-            if "default " not in type_line:
-                after = type_line.split("default")[1]
-                found = "default" + after
-                should = "default [VALUE]"
-                errors.append(
-                    (
-                        "TE01",
-                        MODIN_ERROR_CODES["TE01"].format(
-                            parameter=parameter, should=should, found=found
-                        ),
-                    )
+    for parameter in optional_args:
+        type_line = doc.doc_parameters[parameter][0]
+        if "default: " not in type_line or "optional" in type_line:
+            errors.append(
+                (
+                    "MD01",
+                    MODIN_ERROR_CODES["MD01"].format(
+                        parameter=parameter,
+                        found=type_line,
+                    ),
                 )
+            )
+    return errors
+
+
+def check_spelling_words(doc: Docstring) -> list:
+    """
+    Check spelling of chosen words: "Modin", "pandas", "NumPy" in doc.
+
+    Parameters
+    ----------
+    doc : numpydoc.validate.Docstring
+        Docstring handler.
+
+    Returns
+    -------
+    list
+        List of tuples with Modin error code and its description.
 
+    Notes
+    -----
+    "modin" word is treated as the constant.
+    """
+    if not doc.raw_doc:
+        return []
+    components = set(["Modin", "pandas", "NumPy"])
+    check_words = "|".join(x.lower() for x in components)
+
+    # comments work only with re.VERBOSE
+    pattern = r"""
+    (?:\W|^)                # non-capturing group: [^a-zA-Z0-9_] or start
+    ({check_words})         # words to check, example - "modin|pandas|numpy"
+    (?:                     # non-capturing group
+        [^"\.\/\w\\]        # any symbol except: '"', '.', '\' and any from [a-zA-Z0-9_]
+        | [\.]\s            # or '.' and any whitespace
```

Comment:
```suggestion
        | \.\s            # or '.' and any whitespace
```
I don't get why you'd want to make a "take any of" group with one element inside.

Also does this handle cases like `modin.pandas.DataFrame` correctly?

Reply:
Yes, regexp handle that cases.